db/migrate/20170602111120_add_timestamps_to_documents.rb

```diff
@@ -0,0 +1,110 @@
+class AddTimestampsToDocuments < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+
+  def change
+    add_timestamps(:documents, null: true) unless column_exists?(:documents, :created_at)
+    puts "> Timestamp fields added."
+    puts "> Updating multiple-locale documents without editions..."
+
+    update_multiple_locale_documents_without_editions
+    puts "> Multiple-locale documents without editions updated."
+    puts "> Updating remaining documents..."
+
+    Document.where(created_at: nil).find_each do |document|
+      editions = Edition.where(document_id: document.id).order(:user_facing_version)
+      if editions.exists?
+        update_timestamps_from_editions(document, editions)
+      else
+        update_timestamps_from_events(document)
+      end
```

Comment:
Might be an idea to `puts` something about the progress here - could do a count first and then use `find_each.with_index` to have the number? - just for a bit more feedback while it runs.